### PR TITLE
Io file index key

### PR DESCRIPTION
**Pull request by pvinchon**

### CODE REVIEW

**Review by paucarre — APPROVED** [review]

File: src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua [hunk 1]
```diff
@@ -15,46 +15,36 @@ local redis = require 'redis'
...
 end
 
-function tiefvision_redis_io.last(fileName, data)
-  local keys = tiefvision_redis_io.redisClient:keys("*")
-  local keysInt = {}
+function tiefvision_redis_io.keys(database)
+  local keys = tiefvision_redis_io.redisClient:keys(database .. ":*")
   for i = 1, #keys do
-    keysInt[i] = tonumber(keys[i])
+    keys[i] = string.sub(keys[i], #database + 2)
```
Comment (paucarre) [inline src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua]: where does this +2 come?

File: src/torch/0-tiefvision-commons/tiefvision_commons.lua [hunk 2]
```diff
@@ -85,6 +85,22 @@ function tiefvision_commons.preprocess(im)
   return imageMinusAvg:cuda()
 end
 
+function tiefvision_commons.tableSubtraction(t1, t2)
```
Comment (paucarre) [inline src/torch/0-tiefvision-commons/tiefvision_commons.lua]: Just for your information, it exists https://github.com/torch/tds

**Review by paucarre — APPROVED** [review]
Only a few comments, overall LGTM

File: src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua (hunk 1 above)
Comment (Plippe) [inline src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua]: Sub allows you to discard all characters **before** index. So if you grab the length of a string, you have the last character of that string (+1). Furthermore, I have a `:` between the two strings. To discard it too, I need +1 again (+2).

File: src/torch/0-tiefvision-commons/tiefvision_commons.lua (hunk 2 above)
Comment (Plippe) [inline src/torch/0-tiefvision-commons/tiefvision_commons.lua]: Thanks, didn't know this lib

File: src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua [hunk 3]
```diff
@@ -15,46 +15,36 @@ local redis = require 'redis'
...
-function tiefvision_redis_io.read(fileName)
-  local key = toKey(fileName)
-  local response = tiefvision_redis_io.redisClient:hgetall(key)
-
-  local responseInt = {}
-  for key, value in pairs(response) do
-    local keyInt = tonumber(key)
-    responseInt[keyInt] = value
+function tiefvision_redis_io.read(database, key)
+  local response = tiefvision_redis_io.redisClient:hgetall(database .. ':' .. key)
```
Comment (emersonloureiro) [inline src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua]: This is just for my own curiosity, but why are we overwriting the pairs in `response`. I mean, if that's the response we got from Redis, and we don't seem to be doing anything with the values `v` in the loop below, what are we actually doing by overwriting each key `k`?

Comment (Plippe) [inline src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua]: You are 100% right, bug here

File: src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua (hunk 1 above)
Comment (paucarre) [inline src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua]: Could you maybe add a small comment in the code so next time we go though it we know where it comes from?

File: src/torch/0-tiefvision-commons/tiefvision_commons.lua (hunk 2 above)
Comment (paucarre) [inline src/torch/0-tiefvision-commons/tiefvision_commons.lua]: No need to use it for this PR, just in case you need later on

**Review by emersonloureiro — APPROVED** [review]
Looks ok to me, but I really only glanced over. Not too familiar with this, yet 😄

**Comment by paucarre** [comment]
LGTM

**Pull request merged**